### ISSUE TITLE
data attribute filter accepts date-time

### ISSUE THREAD

**Issue by flinden68** [maykinmedia/objects-api#473]
### Thema / Theme

Objecten API

### Omschrijving / Description

Momenteel kunnen we voor bijvoorbeeld LTE alleen maar een date of numeric waarde gebruiken. 
Maar in ons object hebben we te maken met date-time en dan faalt het LTE filter als je de huidige datum gebruikt om alle objecten op wilt halen voor vandaag en ouder.

### Toegevoegde waarde / Added value

Hierdoor kan je beter filteren op basis van een date-time

**Comment by alextreme** [comment]
JSON doesn't have a native datetime format so this will be quite complicated, we would recommend to store the date separately if it is necessary to filter on it

**Comment by sdegroot** [comment]
@alextreme what do you mean there is no native date time format? If you mean that dates are serialised either as numeric timestamps or string values (RFC 3339 / ISO 8601) then yes.

However, JSON Schema very much allows to enforce those standards upon a property using the "format" attribute. 
See https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-00#rfc.section.7.3

Unfortunately, the validator used in objects-api/objecttypes-api does not adhere to the standards and simply ignores the format field. I discussed this several months ago with @joeribekker but haven't heard back about this. 

@flinden68 is this what you meant?

**Comment by flinden68** [comment]
@sdegroot yes, because we pass on the date also a date format as string, should be that hard also accept a date-time formatted string

**Comment by sdegroot** [comment]
Ah, so it is optional. I would highly suggest enabling this as it will increase data quality -a lot- :)

**Comment by JanBrek** [comment]
@alextreme I agree with @sdegroot. It would help a lot for connecting applications as they would get errors while unmarshalling the json objects. 
We currently have an issue where an object has a wrong date '000' even though the json schema has the date-time format. This causes the search function for that object type to break (One item in the list is invalid, thus the whole list is unable to unmarshall resulting in a broken query/call. 
If the objecten API would honor the dateTime format this error would be caught while trying to store the object.
Must we make a separate (taiga?) issue to request this?

**Comment by alextreme** [comment]
@JanBrek please indeed create a Taiga issue with the details of the object and json schema involved, combined with the API request which failed. I'll then pick this up with Team Bron to triage further

**Comment by flinden68** [comment]
@JanBrek @alextreme an example could be /api/v2/objects?page=1&pageSize=10&type=https://example.com/api/v1/objecttypes/78731088-430f-49fd-9a4c-80ddd42ded28&data_attrs=identificatie__type__exact__bsn,identificatie__value__exact__999990755,publicatiedatum__lte__2025-01-10T12:00:00.000

Nice to would be where the Zone is included

/api/v2/objects?page=1&pageSize=10&type=https://example.com/api/v1/objecttypes/78731088-430f-49fd-9a4c-80ddd42ded28&data_attrs=identificatie__type__exact__bsn,identificatie__value__exact__999990755,publicatiedatum__lte__2025-01-10T12:00:00.000Z

**Comment by PeterVanBragt** [comment]
For now no budget for this feature at GDH (also no internal sponsor).

**Comment by PeterVanBragt** [comment]
Possibly Peter Arkesteijn will order this functionality (to be charged from his "strippenkaart").

**Comment by alextreme** [comment]
@janmasGDH

**Comment by alextreme** [comment]
Pushing this issue back to triage as the scope is being expanded since november and funding is no longer clear. And the questions @JanBrek raises seem important enough to revisit this issue during refinement
